kubetest2/kubetest2-gce/deployer/dumplogs.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog"
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) DumpClusterLogs() error {
+	klog.Info("GCE deployer starting DumpClusterLogs()")
+
+	if err := d.verifyFlags(); err != nil {
+		return fmt.Errorf("dump cluster logs could not verify flags: %s", err)
+	}
+
+	if err := d.makeLogsDir(); err != nil {
+		return fmt.Errorf("couldn't make logs dir: %s", err)
+	}
+
+	// Run sshDump before kubectlDump because kubectl could fail if master kube
+	// didn't come up successfully but the instances were still created and setup
+	// proceeded to a certain point. This allows retrieval of logs that could
+	// indicate why master coming up failed.
+	if err := d.sshDump(); err != nil {
+		return fmt.Errorf("failed to dump logs from instance log files: %s", err)
+	}
+
+	if err := d.kubectlDump(); err != nil {
+		return fmt.Errorf("failed to dump cluster info with kubectl: %s", err)
+	}
+
+	return nil
+}
+
+func (d *deployer) makeLogsDir() error {
+	_, err := os.Stat(d.logsDir)
+
+	if err == nil {
+		return fmt.Errorf("cluster logs directory %s already exists, please clean up manually before continuing", d.logsDir)
+	} else if os.IsNotExist(err) {
+		err := os.Mkdir(d.logsDir, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("failed to create %s: %s", d.logsDir, err)
+		}
+		return nil
+	}
+
+	return fmt.Errorf("unexpected exception when making cluster logs directory: %s", err)
+}
+
+func (d *deployer) sshDump() error {
+	env := d.buildEnv()
+	env = append(env, "KUBE_GCE_INSTANCE_PREFIX=kubernetes")
```

Comment:
clarifying comment would be helpful here.
also kubernetes seems to be the default, do we want to set something up specially for kubetest2? that our Up() will do.

Reply:
Set in buildEnv with comment. Now it will be consistent across all commands. Explained in comment.